### DISCUSSION TITLE
bagaimana cara membuat engine seperti fyp di tiktok

### DISCUSSION THREAD

**Discussion by fatwa13** [teamalgoritma/community#219] [Q&A] [answered]
Selamat siang bapak/ ibu , mohon maaf mengganggu waktu nya . saya ingin bertanya bagaimana cara membuat engine algoritma seperti fyp di tiktok dan tools / method yang apa saja yang harus di pakai terima kasih

**Comment by idoali** [comment, marked as answer]
TikTok memberikan rekomendasi berdasarkan video-video yang pernah kita tonton sebelumnya pak. Mereka memiliki daftar lengkapnya dan membuat rekomendasi menggunakan itu. Algoritma yang biasanya dipakai adalah ALS (*Alternation of Least Squares*) dan *Cosine Similarity*. 

Dua algoritma ini memiliki kemampuan untuk mencari video baru yang mirip dengan video-video yang pernah kita tonton sebelumnya. Misalkan kita suka menonton video viral terkait mobil, maka nanti TikTok akan merekomendasikan video mobil viral yang belum pernah bapak tonton. Jika tidak ada video mobil yang sedang viral, maka kemungkinan bapak akan menerima rekomendasi video motor yang viral karena video ini adalah video yang paling mendekati selera bapak sesuai history.   

Hal ini dapat dilakukan TikTok karena mereka menggunakan algoritma *Cosine Similarity* atau ALS.

Semoga jawabannya cukup menjawab pak.

**Reply by fatwa13** [reply]
selamat sore pak ido,  terima kasih sudah menjawab pertanyaan saya dan semoga ilmu ini bisa saya implementasikan di tempat kerja saya
